Sources/ScheduledTasks.php

```diff
@@ -1542,4 +1542,196 @@ function scheduled_remove_old_drafts()
 	return true;
 }
 
+/**
+ * Prune log_topics, log_boards & log_mark_boards_read.
+ * For users who haven't been active in a long time, purge these records.
+ * For users who haven't been active in a shorter time, mark boards as read,
+ * pruning log_topics.
+ */
+function scheduled_prune_log_topics()
+{
+	global $smcFunc, $sourcedir, $modSettings;
+
+	if (empty($modSettings['mark_read_max_users']))
+		return true;
+
+	// Convert to timestamps for comparison
+	$markReadCutoff = time() - $modSettings['mark_read_beyond'] * 86400;
+	$cleanupBeyond = time() - $modSettings['mark_read_delete_beyond'] * 86400;
+	$maxMembers = $modSettings['mark_read_max_users'];
+
+	// Try to prevent timeouts
+	@set_time_limit(300);
+	if (function_exists('apache_reset_timeout'))
+		@apache_reset_timeout();
+
+	// Start off by finding the records in log_boards, log_topics & log_mark_read
+	// for users who haven't been around the longest...
+
+	// Find the oldest $maxMembers in log_boards
+	$sql = "SELECT lb.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_boards
+		) lb
+		ON lb.id_member = m.id_member
```

Comment:
I would move the "on" part one tab to the right,
since it's part of the join statement or move a line up with the alias.

Reply:
Will do...